mongo/integration/unified/entity.go

```diff
@@ -23,8 +23,14 @@ import (
 	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
 )
 
-// ErrEntityMapOpen is returned when a slice entity is accessed while the EntityMap is open
-var ErrEntityMapOpen = errors.New("slices cannot be accessed while EntityMap is open")
+var (
+	// ErrEntityMapOpen is returned when a slice entity is accessed while the EntityMap is open
+	ErrEntityMapOpen = errors.New("slices cannot be accessed while EntityMap is open")
+
+	// ErrEntityNotFound is returned when an entity is not found in an
+	// EntityMap hash.
+	ErrEntityNotFound = errors.New("entity not found")
```

Comment:
Is there a use case for adding this error variable? As far as I can see, no code compares against that error.

Reply:
Yes it's used in "newEntityNotFoundError" on L723

Reply:
"newEntityNotfoundError" is used used 25 times between "mongo/integration/unified/entity.go" and "mongo/integration/unified/entity_test.go".